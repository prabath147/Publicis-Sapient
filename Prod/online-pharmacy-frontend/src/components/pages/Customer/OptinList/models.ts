import { AddressInterface } from "../../../ui/forms/AddressForm";
import { PageableResponse } from "../../Admin/storeList/models";
import { ProductMiniDTO } from "../../Cart/models";

export interface OptInType {
  address: AddressInterface;
  deliveryDate: Date | string;
  id: number;
  intervalInDays: number;
  name: string;
  numberOfDeliveries: number;
  repeatOrderItems: ProductMiniDTO[];
  userId: number;
}

// export interface Address {
//   addressId?: number;
//   city: string;
//   country: string;
//   pinCode: string|number;
//   state: string;
//   street: string;
// }

// export interface RepeatOrderItem {
//   price: number;
//   productId?: number;
//   productIdFk: number;
//   quantity: number;
// }

export interface OptInListResponse extends PageableResponse {
  data: OptInType[];
}
