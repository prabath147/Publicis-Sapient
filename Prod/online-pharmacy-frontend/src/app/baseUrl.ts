export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
