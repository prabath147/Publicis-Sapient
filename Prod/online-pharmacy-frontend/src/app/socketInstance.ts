
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);