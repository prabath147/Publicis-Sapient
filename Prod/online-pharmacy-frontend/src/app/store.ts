import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
