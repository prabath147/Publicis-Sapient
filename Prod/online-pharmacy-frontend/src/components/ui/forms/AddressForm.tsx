import { Space, Text, TextInput } from "@mantine/core";
import { UseFormReturnType } from "@mantine/form";

interface addressFormPropType {
    form: UseFormReturnType<any>,
    formFieldName?: string
}
export default function AddressForm({ form, formFieldName = "address" }: addressFormPropType) {
    return (
        <div>
            <Space />
            <Text >Address</Text>
            <TextInput label="Street" placeholder='Street' withAsterisk {...form.getInputProps(formFieldName + '.street')} />
            <TextInput label="City" placeholder='City' withAsterisk {...form.getInputProps(formFieldName + '.city')} />
            <TextInput label="State" placeholder='State' withAsterisk {...form.getInputProps(formFieldName + '.state')} />
            <TextInput label="Country" placeholder='Country' withAsterisk {...form.getInputProps(formFieldName + '.country')} />
            <TextInput type="number" label="Pincode" placeholder="PinCode" withAsterisk {...form.getInputProps(formFieldName + '.pinCode')} />
        </div>
    )
}

export const addressValidation = (): any => {
    return {
        city: value => value.length < 2 ? "This field is required!" : null,
        country: value => value.length < 2 ? "This field is required!" : null,
        pinCode: value => Math.floor(value) < 99999 ? "Valid Pincode is required!" : null,
        state: value => value.length < 2 ? "This field is required!" : null,
        street: value => value.length < 2 ? "This field is required!" : null,
    }
}

export const addressInitialValues = (): AddressInterface => {
    return {
        city: '',
        country: '',
        pinCode: '',
        state: '',
        street: ''
    }
}

export interface AddressInterface {
    addressId?: number;
    city: string,
    country: string,
    pinCode: string,
    state: string,
    street: string,
}

