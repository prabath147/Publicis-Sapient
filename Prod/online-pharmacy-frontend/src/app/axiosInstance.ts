import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
