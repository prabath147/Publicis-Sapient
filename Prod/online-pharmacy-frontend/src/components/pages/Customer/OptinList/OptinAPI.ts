import axiosInstance from "../../../../app/axiosInstance";
import { OptInType } from "./models";

// export const getOptinListAPI = (userid: number) => axiosInstance.get("/order/optin/user/" + userid);

// export const getOptinDetailAPI = (optinId: number) =>
//   axiosInstance.get("api/optin/" + optinId);
export const getOptInByIdAPI = (id: number) =>
  axiosInstance.get("order/optin/" + id);

export const deleteOptInByIdAPI = (id: number) =>
  axiosInstance.delete("order/optin/delete/" + id);

export const optInCreateAPI = (data: OptInType) =>
  axiosInstance.post("order/optin/create-optin", data);

export const optInUpdateAPI = (data: OptInType) =>
  axiosInstance.put("order/optin/update-optin", data);

export const getOptInForUserAPI = (userId: number, pageNumber: number, pageSize: number) =>
  axiosInstance.get(`order/optin/user/${userId}?pageNumber=${pageNumber}&pageSize=${pageSize}`);
