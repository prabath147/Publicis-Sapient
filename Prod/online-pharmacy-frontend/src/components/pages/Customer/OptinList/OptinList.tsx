import { useEffect, useState } from "react";

import { Accordion, Center, Container, Grid, Group, Loader, Pagination, Space, Text, Title, UnstyledButton } from "@mantine/core";
import { showNotification } from "@mantine/notifications";
import { IconEdit, IconPill, IconTrash } from '@tabler/icons';
import { Link } from "react-router-dom";
import { useAppSelector } from "../../../../app/hooks";
import { getUserData } from "../../Auth/Login/UserSlice";
import { OptInType } from "./models";
import { deleteOptInByIdAPI, getOptInForUserAPI } from "./OptinAPI";



export default function OptinList() {
  // const optinList = useAppSelector(getOptInList)
  // const dispatch = useAppDispatch()
  const id = useAppSelector(getUserData).id
  const [activePage, setPage] = useState(1);
  const [pageCount, setPageCount] = useState(1);
  const [loading, setLoading] = useState(true);
  const [optInList, setOptInList] = useState<OptInType[]>([])
  const limit = 5;

  const loadDataAction = () => {
    setLoading(true);
    getOptInForUserAPI(id, activePage - 1, limit)
      .then(response => {
        // console.log(response.data);
        setPageCount(response.data.totalPages);
        setOptInList(response.data.data)
        setLoading(false)


        // dispatch(loadOptIn(response.data.data))
      })
      .catch(err => {
        // console.error(err)
        showNotification({ color: "red", message: "oops, something went wrong" })
        setLoading(false)
        // dispatch(loadOptIn(optinMock))
      })
  };

  useEffect(() => {
    loadDataAction();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activePage]);

  if (loading) return (
    <Center style={{ height: "80%" }}>
      <Loader data-testid="Loading" />
    </Center >
  )

  return (
    <div>
      <Container>
        <Title align="center">
          Opt In List
        </Title>
        <br />

        <Accordion variant="contained">
          {optInList.map((optin: OptInType) => (
            <Accordion.Item value={optin.id.toString()} >
              <Accordion.Control icon={<IconPill size={20} color="green" />}>
                <Group position="apart" spacing="sm">
                  <Title size={16}>{optin.name}</Title>
                  <Text>Next: {optin.deliveryDate.toString()}</Text>
                </Group>
              </Accordion.Control>
              <Accordion.Panel>

                <Group position="right">
                  <Link to={"/optin/update/" + optin.id}>
                    <IconEdit color="green" />
                  </Link>
                  <UnstyledButton onClick={() => {
                    deleteOptInByIdAPI(optin.id)
                      .then(() => {
                        setOptInList(optInList.filter(obj => obj.id !== optin.id))
                      })
                      .catch(() => {
                        showNotification({ color: 'red', message: 'Oops, Something went wrong' })
                      })
                  }}>
                    <IconTrash color="red" />
                  </UnstyledButton>
                </Group>

                <Grid>
                  <Grid.Col span={4}>Id</Grid.Col>
                  <Grid.Col span={8}>{optin.id}</Grid.Col>

                  <Grid.Col span={4}>Name</Grid.Col>
                  <Grid.Col span={8}>{optin.name}</Grid.Col>

                  <Grid.Col span={4}>Delivery Intervals</Grid.Col>
                  <Grid.Col span={8}>{optin.intervalInDays} days</Grid.Col>

                  <Grid.Col span={4}>Next Delivery</Grid.Col>
                  <Grid.Col span={8}>{optin.deliveryDate.toString()} </Grid.Col>

                  <Grid.Col span={4}>Deliveries Left</Grid.Col>
                  <Grid.Col span={8}>{optin.numberOfDeliveries}</Grid.Col>
                </Grid>

                {/* <CartCard cart={optin.cart} /> */}
              </Accordion.Panel>
            </Accordion.Item>
          ))}
        </Accordion>

        <Space h="xl" />

        <Pagination
          position="center"
          page={activePage}
          onChange={setPage}
          total={pageCount}
        />
      </Container>
    </div >
  );
}


