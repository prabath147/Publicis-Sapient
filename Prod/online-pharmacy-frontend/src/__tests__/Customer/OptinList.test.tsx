import { render, waitForElementToBeRemoved } from '@testing-library/react';
import dayjs from 'dayjs';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../../app/store';
import OptinList from '../../components/pages/Customer/OptinList/OptinList';
import { addressInitialValues } from '../../components/ui/forms/AddressForm';
import mockAxios from '../../__mocks__/axios';

const MockOptinList = () => (
    <Provider store={store} >
        <BrowserRouter>
            <OptinList />
        </BrowserRouter>
    </Provider >
)



describe('optinList', () => {

    jest
        .useFakeTimers()
        .setSystemTime(new Date('2020-01-01'));

    afterEach(() => {
        jest.resetAllMocks();
    });

    const data = {
        "data": [
            {
                id: 2,
                intervalInDays: 2,
                name: "fever",
                deliveryDate: dayjs(new Date()).add(3, "days").toDate(),
                numberOfDeliveries: 3,
                userId: 20,
                address: addressInitialValues(),
                repeatOrderItems: []
            },
        ],
        "pageNumber": 0,
        "pageSize": 10,
        "totalRecords": 0,
        "totalPages": 0,
        "isLastPage": true
    }

    it('should render correctly', async () => {
        mockAxios.get.mockImplementationOnce(() =>
            Promise.reject(),
        )

        const { asFragment } = render(<MockOptinList />)
        expect(asFragment()).toMatchSnapshot();
    })

    it('should render with data', async () => {

        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({ data: data }),
        )

        const { asFragment, getByTestId } = render(<MockOptinList />)

        await waitForElementToBeRemoved(getByTestId(/Loading/i));
        // expect(element.length).toEqual(2)

        expect(asFragment()).toMatchSnapshot()

    })
})
