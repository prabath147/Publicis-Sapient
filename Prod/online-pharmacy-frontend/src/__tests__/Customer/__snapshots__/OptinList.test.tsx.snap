// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optinList should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mantine-8od8ev"
    style="height: 80%;"
  >
    <svg
      class="mantine-1avyp1d"
      data-testid="Loading"
      height="36px"
      role="presentation"
      stroke="#228be6"
      viewBox="0 0 38 38"
      width="36px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(2.5 2.5)"
        >
          <circle
            cx="16"
            cy="16"
            r="16"
            stroke-opacity=".5"
          />
          <path
            d="M32 16c0-9.94-8.06-16-16-16"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 16 16"
              repeatCount="indefinite"
              to="360 16 16"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`optinList should render with data 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mantine-Container-root mantine-15pjuqq"
    >
      <h1
        class="mantine-Text-root mantine-Title-root mantine-1l71p2l"
      >
        Opt In List
      </h1>
      <br />
      <div
        class="mantine-1avyp1d"
        data-accordion="true"
      >
        <div
          class="mantine-Accordion-item mantine-oomffu"
        >
          <button
            aria-controls="mantine-r0-panel-2"
            aria-expanded="false"
            class="mantine-UnstyledButton-root mantine-Accordion-control mantine-xizyzk"
            data-accordion-control="true"
            id="mantine-r0-control-2"
            type="button"
          >
            <div
              class="mantine-1h6pkea mantine-Accordion-chevron"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 15 15"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="mantine-pildck mantine-Accordion-label"
            >
              <div
                class="mantine-Group-root mantine-qvrn7b"
              >
                <h1
                  class="mantine-Text-root mantine-Title-root mantine-Group-child mantine-c374lq"
                >
                  fever
                </h1>
                <div
                  class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
                >
                  Next: Sat Jan 04 2020 05:30:00 GMT+0530 (India Standard Time)
                </div>
              </div>
            </div>
            <div
              class="mantine-r9y4tq mantine-Accordion-icon"
            >
              <svg
                class="icon icon-tabler icon-tabler-pill"
                fill="none"
                height="20"
                stroke="green"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M4.5 12.5l8 -8a4.94 4.94 0 0 1 7 7l-8 8a4.94 4.94 0 0 1 -7 -7"
                />
                <line
                  x1="8.5"
                  x2="15.5"
                  y1="8.5"
                  y2="15.5"
                />
              </svg>
            </div>
          </button>
          <div
            aria-hidden="true"
            aria-labelledby="mantine-r0-control-2"
            class="mantine-Accordion-panel mantine-1supnlp"
            id="mantine-r0-panel-2"
            role="region"
            style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
          >
            <div
              style="opacity: 0; transition: opacity 200ms ease;"
            >
              <div
                class="mantine-ukbbnm mantine-Accordion-content"
              >
                <div
                  class="mantine-Group-root mantine-147cgkf"
                >
                  <a
                    class="mantine-gnzaph mantine-Group-child"
                    href="/optin/update/2"
                  >
                    <svg
                      class="icon icon-tabler icon-tabler-edit"
                      fill="none"
                      height="24"
                      stroke="green"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <path
                        d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                      />
                      <path
                        d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                      />
                      <path
                        d="M16 5l3 3"
                      />
                    </svg>
                  </a>
                  <button
                    class="mantine-UnstyledButton-root mantine-Group-child mantine-14e47jh"
                    type="button"
                  >
                    <svg
                      class="icon icon-tabler icon-tabler-trash"
                      fill="none"
                      height="24"
                      stroke="red"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                        stroke="none"
                      />
                      <line
                        x1="4"
                        x2="20"
                        y1="7"
                        y2="7"
                      />
                      <line
                        x1="10"
                        x2="10"
                        y1="11"
                        y2="17"
                      />
                      <line
                        x1="14"
                        x2="14"
                        y1="11"
                        y2="17"
                      />
                      <path
                        d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                      />
                      <path
                        d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="mantine-Grid-root mantine-nsdj"
                >
                  <div
                    class="mantine-Col-root mantine-ue7w1"
                  >
                    Id
                  </div>
                  <div
                    class="mantine-Col-root mantine-194xbhl"
                  >
                    2
                  </div>
                  <div
                    class="mantine-Col-root mantine-ue7w1"
                  >
                    Name
                  </div>
                  <div
                    class="mantine-Col-root mantine-194xbhl"
                  >
                    fever
                  </div>
                  <div
                    class="mantine-Col-root mantine-ue7w1"
                  >
                    Delivery Intervals
                  </div>
                  <div
                    class="mantine-Col-root mantine-194xbhl"
                  >
                    2 days
                  </div>
                  <div
                    class="mantine-Col-root mantine-ue7w1"
                  >
                    Next Delivery
                  </div>
                  <div
                    class="mantine-Col-root mantine-194xbhl"
                  >
                    Sat Jan 04 2020 05:30:00 GMT+0530 (India Standard Time) 
                  </div>
                  <div
                    class="mantine-Col-root mantine-ue7w1"
                  >
                    Deliveries Left
                  </div>
                  <div
                    class="mantine-Col-root mantine-194xbhl"
                  >
                    3
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mantine-1eabysl"
      />
      <div
        class="mantine-Group-root mantine-1q3od06"
        role="navigation"
      >
        <button
          aria-disabled="true"
          class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
          disabled=""
          type="button"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          aria-disabled="false"
          class="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
          type="button"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
